zenoh/src/publication.rs

```diff
@@ -293,25 +406,22 @@ impl<'a> Publisher<'a> {
 
     /// Change the `congestion_control` to apply when routing the data.
     #[inline]
-    pub fn congestion_control(mut self, congestion_control: CongestionControl) -> Self {
+    pub fn set_congestion_control(&mut self, congestion_control: CongestionControl) {
```

Comment:
Why the `set_`?

Reply:
Because these methods are not builder ones, but setters for object itself. Otherwise it's unclear, do they set or return value

Reply:
Do we really want to allow override of values once the publisher is already created? I'm skeptical about it because it allows the user to override at a later stage what has been already configured with the `declare_publisher`. I'm more inclined to not having such `set_` functions.

Reply:
I think If this functionality is not really useful, better to remove it to give us more freedom for future updates. So I don't mind to remove it.
In C API it's possible to only set options on declaration stage, so this change will not affect the C/C++ bindings